src/pudl/transform/eia176.py

```diff
@@ -0,0 +1,76 @@
+"""Module to perform data cleaning functions on EIA176 data tables."""
+
+import warnings
+
+import pandas as pd
+from dagster import ExperimentalWarning, asset, asset_check
+
+from pudl.logging_helpers import get_logger
+
+logger = get_logger(__name__)
+
+# Asset Checks are still Experimental, silence the warning since we use them
+# everywhere.
+warnings.filterwarnings("ignore", category=ExperimentalWarning)
+
+
+@asset
+def _core_eia176__data(raw_eia176__data: pd.DataFrame) -> pd.DataFrame:
+    """Take entity-attribute-value rows and convert to rows with primary key and one column per variable."""
+    raw_eia176__data["variable_name"] = (
+        raw_eia176__data["line"] + "_" + raw_eia176__data["atype"]
+    )
+    primary_key = ["report_year", "area", "id"]
+    variable_names = list(raw_eia176__data.groupby("variable_name").count().index)
+    wide_table = pd.DataFrame(columns=primary_key + variable_names)
+
+    granular_data = raw_eia176__data[
+        (raw_eia176__data["company"] != " Total of All Companies")
+    ]
+    for report_year, area, id_ in granular_data.groupby(primary_key).count().index:
```

Comment:
It looks to me like you're trying to manually do an [unstack()](https://pandas.pydata.org/pandas-docs/version/0.16.1/generated/pandas.DataFrame.unstack.html) operation (take the variable names in the `"variable_name"` column and turn them into a bunch of columns). There's 51k PK groups here, and doing `iterrows()` across each group and recombining is pretty expensive in pandas.

I was able to get a similarly shaped result across *all* years in ~5s by leveraging the built-in `unstack` functionality and some inelegant cleanup of `unstack` artifacts.

```python
primary_key = ["report_year", "area", "id"]
raw_eia176__data["variable_name"] = (
    raw_eia176__data["line"] + "_" + raw_eia176__data["atype"]
)
# TODO should probably sanitize this company name somewhere beforehand
granular = raw_eia176__data.loc[
    raw_eia176__data.company.str.strip().str.lower() != "total of all companies"
]
unstacked = (
    granular
    .drop(columns=["itemsort", "item", "atype", "line", "company"])
    .set_index(primary_key + ["variable_name"])
    .unstack(level="variable_name")
)
# columns is a weird multi-index with ("value", "actual column name") - clean that up
unstacked.columns = unstacked.columns.droplevel(0)
unstacked.columns.name = None # gets rid of "variable_name" name of columns index

# TODO instead of "first NA value we see in each column" applied willy-nilly, we could check to see if there are any conflicting non-null values using .count() first.
condensed = unstacked.groupby(level=primary_key).first().reset_index()
return condensed
```

One sort of weird thing that I'm curious about - in #3501 it looks like we wanted to use `item` to identify variables - should we be doing that instead of the `line` + `atype` thing we have going on here?

Reply:
Outputs for comparison/posterity:
<details>
<summary>`unstack`, all years</summary>

```
      report_year     area          id      1010_CS   1010_CT     1010_VL     101T_VL     1020_CS  1020_CT    1020_VL     102T_VL     1030_CS  1030_CT    1030_VL     103T_VL  1040_CS  1040_CT  1040_VL    104T_VL  1050_CS  1050_CT  1050_VL  105T_VL  1060_CS  1060_CT  1060_VL  106T_VL  110_VL  1110_CS  1110_CT  1110_VL  1120_CS  1120_CT     1120_VL   1130_CS  1130_CT     1130_VL  1140_CS  1140_CT    1140_VL  1150_CS  1150_CT  1150_VL  1160_CT  1160_VL  120_VL   1210_VL  1220_VL   1230_VL  1240_VL  1250_VL  1260_VL  1310_VL   1320_VL  1330_VL      1400_VL  1500_VL  1600_VL  1700_VL     1810_VL     1820_VL    1830_VL  1840_VL      1900_VL  2000_VL  210_VL    220_VL  230_VL      300_VL  3014_CT  3014_VL  3_VL  400_VL  410_CS    410_VL  420_VL       500_VL  5_CT  600_VL  6_CT       700_VL  800_VL  810_VL  810_YA  820_VL  820_YA  900_VL  9_VL
0            1997  Alabama  17600048AL     251371.0     625.0     29469.0     29469.0     69956.0     50.0     8358.0      8358.0         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN      37827.0  -9795.0     NaN       NaN     NaN     28032.0      NaN      NaN   NaN     NaN     NaN       NaN     NaN          NaN   NaN     NaN   NaN      28032.0     NaN     NaN     NaN     NaN     NaN  1100.0   NaN
1            1997  Alabama  17600049AL     221993.0     501.0     28016.0     28016.0     27135.0     22.0     4606.0      4606.0         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN      32622.0   1417.0     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN      34039.0   NaN     NaN   NaN      34039.0     NaN     NaN     NaN     NaN     NaN  1018.0   NaN
2            1997  Alabama  17600139AL  246105034.0  423130.0  29308019.0  29308019.0  72340233.0  35652.0  9951817.0  20985483.0  16947052.0   1498.0  2724147.0  51277620.0      NaN      NaN      NaN  7028200.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      0.0     87.0  11033666.0       0.0    194.0  48553473.0      0.0      3.0  7028200.0      NaN      NaN      NaN      NaN      NaN     NaN  179045.0      NaN       NaN      NaN      NaN      NaN      NaN  783604.0      NaN          NaN      NaN      NaN      NaN         NaN         NaN  7800144.0      NaN  117362115.0  49682.0     NaN  404450.0     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN  117007347.0   NaN     NaN   NaN  117411797.0     NaN     NaN     NaN     NaN     NaN  1017.0   NaN
3            1997  Alabama  17600141AL          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN  14356530.0      NaN      NaN      NaN  1598402.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       0.0      9.0  14356530.0      0.0      1.0  1598402.0      NaN      NaN      NaN      NaN      NaN     NaN   52657.0      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN  18275351.0         NaN        NaN      NaN   34282940.0  80555.0     NaN       NaN     NaN  10785270.0      NaN      NaN   NaN     NaN     NaN       NaN     NaN   23578225.0   NaN     NaN   NaN   34363495.0     NaN     NaN     NaN     NaN     NaN     NaN   NaN
4            1997  Alabama  17600162AL    1344834.0    3488.0    204256.0    204256.0    842826.0    523.0   135980.0    135980.0   2427378.0      7.0   703608.0    703608.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN    1043844.0    165.0     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN    1044009.0   NaN     NaN   NaN    1044009.0     NaN     NaN     NaN     NaN     NaN  1037.0   NaN
...           ...      ...         ...          ...       ...         ...         ...         ...      ...        ...         ...         ...      ...        ...         ...      ...      ...      ...        ...      ...      ...      ...      ...      ...      ...      ...      ...     ...      ...      ...      ...      ...      ...         ...       ...      ...         ...      ...      ...        ...      ...      ...      ...      ...      ...     ...       ...      ...       ...      ...      ...      ...      ...       ...      ...          ...      ...      ...      ...         ...         ...        ...      ...          ...      ...     ...       ...     ...         ...      ...      ...   ...     ...     ...       ...     ...          ...   ...     ...   ...          ...     ...     ...     ...     ...     ...     ...   ...
51365        2022  Wyoming  17678821WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN   1926358.0      6.0   228238.0    228238.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN  721281.0      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN  13711595.0        NaN      NaN   14661114.0      0.0     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN   14661114.0   NaN     NaN   NaN   14661114.0     NaN     NaN     NaN     NaN     NaN  1070.0   NaN
51366        2022  Wyoming  17695156WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN   1851326.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN   76757.0      1.0   1851326.0      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN   1522.0    460891.0         NaN        NaN      NaN    2313739.0  -1522.0     NaN       NaN     NaN   2312217.0      NaN      NaN   NaN     NaN     NaN       NaN     NaN          NaN   NaN     NaN   NaN    2312217.0     NaN     NaN     NaN     NaN     NaN  1000.0   NaN
51367        2022  Wyoming  17695458WY    1588901.0    1550.0    134273.0    134273.0   1426548.0    277.0   125426.0    125426.0         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN  19892.0      1.0   1819.0   1819.0      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN    1530.0      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN    215.0         NaN         NaN        NaN      NaN     263263.0   2823.0     NaN       NaN     NaN         NaN      NaN      2.0   NaN     NaN     NaN  266086.0     NaN          NaN   NaN     NaN   NaN     266086.0     NaN     NaN     NaN     NaN     NaN  1095.0   NaN
51368        2022  Wyoming  17696301WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN   5450009.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN  234285.0      2.0   5450009.0      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN    5450009.0      NaN     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN    5450009.0   NaN     NaN   NaN    5450009.0     NaN     NaN     NaN     NaN     NaN  1050.0   NaN
51369        2022  Wyoming  17699910WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN  120000000.0      NaN      NaN      NaN         NaN         NaN        NaN      NaN          NaN      NaN     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN          NaN   NaN     NaN   NaN          NaN     NaN     NaN     NaN     NaN     NaN     NaN   NaN

[51370 rows x 88 columns]
```
</details>

<details><summary>existing code, all years</summary>

```
      report_year     area          id      1010_CS   1010_CT     1010_VL     101T_VL     1020_CS  1020_CT    1020_VL     102T_VL     1030_CS  1030_CT    1030_VL     103T_VL  1040_CS  1040_CT  1040_VL    104T_VL  1050_CS  1050_CT  1050_VL  105T_VL  1060_CS  1060_CT  1060_VL  106T_VL  110_VL  1110_CS  1110_CT  1110_VL  1120_CS  1120_CT     1120_VL   1130_CS  1130_CT     1130_VL  1140_CS  1140_CT    1140_VL  1150_CS  1150_CT  1150_VL  1160_CT  1160_VL  120_VL   1210_VL  1220_VL   1230_VL  1240_VL  1250_VL  1260_VL  1310_VL   1320_VL  1330_VL      1400_VL  1500_VL  1600_VL  1700_VL     1810_VL     1820_VL    1830_VL  1840_VL      1900_VL  2000_VL  210_VL    220_VL  230_VL      300_VL  3014_CT  3014_VL  3_VL  400_VL  410_CS    410_VL  420_VL       500_VL  5_CT  600_VL  6_CT       700_VL  800_VL  810_VL  810_YA  820_VL  820_YA  900_VL  9_VL
0            1997  Alabama  17600048AL     251371.0     625.0     29469.0     29469.0     69956.0     50.0     8358.0      8358.0         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN      37827.0  -9795.0     NaN       NaN     NaN     28032.0      NaN      NaN   NaN     NaN     NaN       NaN     NaN          NaN   NaN     NaN   NaN      28032.0     NaN     NaN     NaN     NaN     NaN  1100.0   NaN
1            1997  Alabama  17600049AL     221993.0     501.0     28016.0     28016.0     27135.0     22.0     4606.0      4606.0         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN      32622.0   1417.0     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN      34039.0   NaN     NaN   NaN      34039.0     NaN     NaN     NaN     NaN     NaN  1018.0   NaN
2            1997  Alabama  17600139AL  246105034.0  423130.0  29308019.0  29308019.0  72340233.0  35652.0  9951817.0  20985483.0  16947052.0   1498.0  2724147.0  51277620.0      NaN      NaN      NaN  7028200.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      0.0     87.0  11033666.0       0.0    194.0  48553473.0      0.0      3.0  7028200.0      NaN      NaN      NaN      NaN      NaN     NaN  179045.0      NaN       NaN      NaN      NaN      NaN      NaN  783604.0      NaN          NaN      NaN      NaN      NaN         NaN         NaN  7800144.0      NaN  117362115.0  49682.0     NaN  404450.0     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN  117007347.0   NaN     NaN   NaN  117411797.0     NaN     NaN     NaN     NaN     NaN  1017.0   NaN
3            1997  Alabama  17600141AL          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN  14356530.0      NaN      NaN      NaN  1598402.0      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       0.0      9.0  14356530.0      0.0      1.0  1598402.0      NaN      NaN      NaN      NaN      NaN     NaN   52657.0      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN  18275351.0         NaN        NaN      NaN   34282940.0  80555.0     NaN       NaN     NaN  10785270.0      NaN      NaN   NaN     NaN     NaN       NaN     NaN   23578225.0   NaN     NaN   NaN   34363495.0     NaN     NaN     NaN     NaN     NaN     NaN   NaN
4            1997  Alabama  17600162AL    1344834.0    3488.0    204256.0    204256.0    842826.0    523.0   135980.0    135980.0   2427378.0      7.0   703608.0    703608.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN    1043844.0    165.0     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN    1044009.0   NaN     NaN   NaN    1044009.0     NaN     NaN     NaN     NaN     NaN  1037.0   NaN
...           ...      ...         ...          ...       ...         ...         ...         ...      ...        ...         ...         ...      ...        ...         ...      ...      ...      ...        ...      ...      ...      ...      ...      ...      ...      ...      ...     ...      ...      ...      ...      ...      ...         ...       ...      ...         ...      ...      ...        ...      ...      ...      ...      ...      ...     ...       ...      ...       ...      ...      ...      ...      ...       ...      ...          ...      ...      ...      ...         ...         ...        ...      ...          ...      ...     ...       ...     ...         ...      ...      ...   ...     ...     ...       ...     ...          ...   ...     ...   ...          ...     ...     ...     ...     ...     ...     ...   ...
51365        2022  Wyoming  17678821WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN   1926358.0      6.0   228238.0    228238.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN  721281.0      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN  13711595.0        NaN      NaN   14661114.0      0.0     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN   14661114.0   NaN     NaN   NaN   14661114.0     NaN     NaN     NaN     NaN     NaN  1070.0   NaN
51366        2022  Wyoming  17695156WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN   1851326.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN   76757.0      1.0   1851326.0      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN   1522.0    460891.0         NaN        NaN      NaN    2313739.0  -1522.0     NaN       NaN     NaN   2312217.0      NaN      NaN   NaN     NaN     NaN       NaN     NaN          NaN   NaN     NaN   NaN    2312217.0     NaN     NaN     NaN     NaN     NaN  1000.0   NaN
51367        2022  Wyoming  17695458WY    1588901.0    1550.0    134273.0    134273.0   1426548.0    277.0   125426.0    125426.0         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN  19892.0      1.0   1819.0   1819.0      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN    1530.0      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN    215.0         NaN         NaN        NaN      NaN     263263.0   2823.0     NaN       NaN     NaN         NaN      NaN      2.0   NaN     NaN     NaN  266086.0     NaN          NaN   NaN     NaN   NaN     266086.0     NaN     NaN     NaN     NaN     NaN  1095.0   NaN
51368        2022  Wyoming  17696301WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN   5450009.0      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN  234285.0      2.0   5450009.0      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN          NaN      NaN      NaN      NaN         NaN         NaN        NaN      NaN    5450009.0      NaN     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN    5450009.0   NaN     NaN   NaN    5450009.0     NaN     NaN     NaN     NaN     NaN  1050.0   NaN
51369        2022  Wyoming  17699910WY          NaN       NaN         NaN         NaN         NaN      NaN        NaN         NaN         NaN      NaN        NaN         NaN      NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN      NaN      NaN      NaN      NaN      NaN         NaN       NaN      NaN         NaN      NaN      NaN        NaN      NaN      NaN      NaN      NaN      NaN     NaN       NaN      NaN       NaN      NaN      NaN      NaN      NaN       NaN      NaN  120000000.0      NaN      NaN      NaN         NaN         NaN        NaN      NaN          NaN      NaN     NaN       NaN     NaN         NaN      NaN      NaN   NaN     NaN     NaN       NaN     NaN          NaN   NaN     NaN   NaN          NaN     NaN     NaN     NaN     NaN     NaN     NaN   NaN

[51370 rows x 88 columns]
```
</details>

They seem to be the same if you `df.compare()` the two as well.

Reply:
Thanks. I was looking for a built-in function and it wasn't a simple transpose. I've worked this in but need to get the pre-commit hooks passing. Will chip away at this over the next few days.